finality-provider/service/fp_instance.go

```diff
@@ -724,12 +733,60 @@ func (fp *FinalityProviderInstance) CommitPubRand(targetBlockHeight uint64) (*ty
 
 	// Update metrics
 	fp.metrics.RecordFpRandomnessTime(fp.GetBtcPkHex())
-	fp.metrics.RecordFpLastCommittedRandomnessHeight(fp.GetBtcPkHex(), lastCommittedHeight)
 	fp.metrics.AddToFpTotalCommittedRandomness(fp.GetBtcPkHex(), float64(len(pubRandList)))
 
 	return res, nil
 }
 
+// TestCommitPubRand is exposed for devops/testing purpose to allow manual committing public randomness in cases
+// where FP is stuck due to lack of public randomness (https://github.com/babylonlabs-io/finality-provider/issues/115).
+//
+// Note:
+// - this function is similar to `CommitPubRand` but should not be used in the main pubrand submission loop.
+// - it will always start from the last committed height + 1
+// - if targetBlockHeight is too large, it will commit multiple fp.cfg.NumPubRand pairs in a loop until reaching the targetBlockHeight
+func (fp *FinalityProviderInstance) TestCommitPubRand(targetBlockHeight uint64) error {
+	lastCommittedHeight, err := fp.GetLastCommittedHeight()
+	if err != nil {
+		return err
+	}
+	fp.metrics.RecordFpLastCommittedRandomnessHeight(fp.GetBtcPkHex(), lastCommittedHeight)
```

Comment:
seems more accurate to record the local last committed height after the commit is made